tokio-epoll-uring/src/system/kernel_support.rs

```diff
@@ -0,0 +1,56 @@
+use std::{io::Write, os::unix::process::CommandExt};
+
+/// Check whether a [`crate::System`]'s consumes memlock rlimit budget.
+///
+/// Unsafe because this function forks & uses [`std::os::unix::process::Command::pre_exec`].
+///
+/// Panics in a bunch of places.
+///
+pub unsafe fn does_system_consume_memlock_rlimit_budget() -> bool {
+    // Fork a child process.
+    // In the child process, set up a tokio-epoll-uring system.
+    // Compare the memlock rlimit usage before and after.
+
+    // poor man's bincode
+    #[repr(C, packed)]
+    struct RlimitPrePost {
+        pre: (u64, u64),
+        post: (u64, u64),
+    }
+    // Ensure `true` binary exists.
+    std::process::Command::new("true")
+        .output()
+        .expect("canont exec `true` binary");
+    let output = std::process::Command::new("true")
+        .pre_exec(|| {
+            let rt = tokio::runtime::Builder::new_current_thread()
+                .enable_all()
+                .build()?;
+            let pre = nix::sys::resource::getrlimit(nix::sys::resource::Resource::RLIMIT_MEMLOCK)
+                .unwrap();
+            let system = rt.block_on(crate::System::launch()).unwrap();
+            // use the memory
+            let ((), res) = rt.block_on(system.nop());
+            res.unwrap();
+            let post = nix::sys::resource::getrlimit(nix::sys::resource::Resource::RLIMIT_MEMLOCK)
+                .unwrap();
```

Comment:
This doesn't make sense. What we want is not to get the limit, but the _usage_.

There's a file in `procfs` for that, the test I added in an earlier commit uses it.